### ISSUE TITLE
Pass the Dimming transition duration parameter to zwave dimmer device

### ISSUE THREAD

**Issue by rezzalex** [domoticz/domoticz#5179]
Hello,

after moving to ZwavejstoMQTT, I saw that my Qubino mini dimmer was quite slow to operate and report its status.
This specific device behavior is explained here:  https://github.com/zwave-js/zwavejs2mqtt/issues/2226#

It can be compensated by integrating the Transition Duration parameter directly from Domoticz, but I did not find amy way to do that.
![image](https://user-images.githubusercontent.com/35136439/156346040-1eb088e0-3b6c-4ad1-8ce2-9f694d0d04b0.png)

Would it require a specific new integration?

thanks a lot

**Comment by gizmocuz** [comment]
By slow you mean the transition is slow ? And specifying 0 for the transition duration would set it to the wanted dim level straight away ?
Domoticz is using the HASS MQTT Auto Discovery protocol, would be enough to also send

"transition": 0

when setting the dim state ?

Maybe you can try this with MQTT Explorer ?

**Comment by rezzalex** [comment]
`By slow you mean the transition is slow? And specifying 0 for the transition duration would set it to the wanted dim level straight away ?`

as explained in the linked post, for this device, by default, when actioning the device from the Domoticz gateway, a parameter "time to dim from 0 to 100" is applied. Normally, this parameter is used to determine how long you have to push the physical switch to dim from 0 to 100. It is only when the dim action is over that the device report back its dim status. So, it is quite long...

Using the transition duration parameter allow bypassing this "physical" duration with the one specified

`Maybe you can try this with MQTT Explorer ?`

Not sure, in the Zwavejs2MQTT screen capture above, you can see that the transition duration does not seem to be a configuration parameter we can pass, there is even no "send" button in the interface, just writing "1s" is enough for the parameter to be taken into account.

Is it possible to search for inspiration in what has been done for Home Assisstant? (see the post link past previously

**Comment by gizmocuz** [comment]
@rezzalex , maybe you can ask in the HASS ZWaveJS2MQTT issue what we need to send when using the HASS MQTT Auto Discovery method, if it is just adding a additional "transition" field with a value of 0 then we can add this.
Or you could ask if we do not specify this parameter that this is done internally in ZWaveJS2MQTT ?

**Comment by rezzalex** [comment]
in the Hass issue, they describe how it can be easily implemented directly into HASS.
Also, how it works from a Zwavejs2Mqtt point of view : https://github.com/zwave-js/zwavejs2mqtt/issues/2226#issuecomment-1026998619

I am afraid any other "technical translation" would be beyond my knowledge...

**Comment by waltervl** [comment]
Just my 2 cents: capture the MQTT message with MQTT Explorer when you set the transition to 0s through the ZwaveJS2MQTT User Interface
In HA default transition can be set by a profile file https://github.com/zwave-js/zwavejs2mqtt/issues/2226#issuecomment-1029937186

**Comment by rezzalex** [comment]
in Mqtt explorer, I have tested with and without a value like "1s" or "#s" filled in "transition duration" field in ZwaveJs2mqtt control panel.
The result is the same, only the parameter 38 "targetValue" is changed. IT seems there is no zwave command in zwaje2mqtt for this parameter.
but Qubino said `The raw command for that if you want to test it is:
26 01 63 00 the value 63 is the dimming value and the 00 is the speed (all values are in hexadecimal)
​
0x26 COMMAND_CLASS_SWITCH_MULTILEVEL
0x01 SWITCH_MULTILEVEL_SET
0x63 Value (
0x00 Dimming Duration`
But as You said @waltervl , it has still been implemented in HASS

**Comment by waltervl** [comment]
But then HA has to send an MQTT message to ZwaveJS2MQTT including this transition duration parameter to switch the bulb. The question is how that MQTT message should look like.

**Comment by rezzalex** [comment]
@waltervl , quoting [AlCalzone](https://github.com/AlCalzone) on the zwavejs2mqtt github, 
"
`
Variant a)
setValue for targetValue with the options parameter (https://zwave-js.github.io/node-zwave-js/#/api/node?id=setvalue), where options should contain the transitionDuration field:
https://github.com/zwave-js/node-zwave-js/blob/deee3107e76b03edc09fb3f472ca0a90fd9cfd60/packages/core/src/values/Metadata.ts#L46-L47

TL;DR this should do the trick:

node.setValue(
  {
    commandClass: 0x26,
    property: "targetValue",
  },
  99, // (= 0x63) or any other target value
  { transitionDuration: "default" }
);
@robertsLando It would be a good idea to expose these options via the UI in general - I think HomeAssistant already does.
Could also define defaults for the setValue options in general or per node.

Variant b)
Using the sendCommand MQTT API, invoke [this method](https://zwave-js.github.io/node-zwave-js/#/api/CCs/MultilevelSwitch?id=set) with the parameters [99, "default"]
`
"

I don't know how to use mqtt explorer to send this. Is it enought to integrate this into MQTT auto discovery plugin ?

**Comment by gizmocuz** [comment]
I think it is a good idea to expose these values in ZWaveJS2MQTT indeed so you can setup your zwave devices directly without a third party application

**Comment by robertsLando** [comment]
![Schermata da 2022-03-04 13-50-01](https://user-images.githubusercontent.com/11502495/156766547-9c3f3067-f4c8-422c-9d84-8dd65732f618.png)


By filling those fields them will be added to the set value call

**Comment by gizmocuz** [comment]
Thats great! Thank you!

**Comment by rezzalex** [comment]
`By filling those fields them will be added to the set value call`

from ZWaveJS2MQTT control panel, yes, from Domoticz, no

**Comment by robertsLando** [comment]
Cannot speak for Domoticz I just answered as a user has tagged me here :)

**Comment by gizmocuz** [comment]
@rezzalex , this is a one time setting I hope ?

**Comment by rezzalex** [comment]
I don't know

**Comment by gizmocuz** [comment]
Best to test...

**Comment by rezzalex** [comment]
the value for the field "transition duration" have to be filled in ZWaveJS2MQTT control panel to be taken into account, if removed (so blank, nothing) the default value for dimming is applied

**Comment by rezzalex** [comment]
> But then HA has to send an MQTT message to ZwaveJS2MQTT including this transition duration parameter to switch the bulb. The question is how that MQTT message should look like.

Not supported with mqtt set, but another API can be used like "setValue" or "sendcommand" , but maybe it is better to wait for the "natural" implementation in ZwaveJstoMQTT ?

**Comment by gizmocuz** [comment]
Yes it would be great if they implemented all configuration also natively in ZWaveJS2MQTT

**Comment by robertsLando** [comment]
https://github.com/zwave-js/zwavejs2mqtt/pull/2453

**Comment by rezzalex** [comment]
Hello @gizmocuz ,

the features have been implemented on Zwavejs2MQTT side : 

https://github.com/zwave-js/zwavejs2mqtt/commit/cbf57c4ecfb341891bee65ce258013975d16b348
Is it now possible to pass options via MQTT messages.

Do you think in DZ, a new UI that would allow passing special parameters should be added?
Addin a new paramter for each "dimmer" device ?

![image](https://user-images.githubusercontent.com/35136439/173107568-8227dd1e-3936-4251-946d-32a40073f592.png)

in HA, it seems they can specify that for all light : https://www.home-assistant.io/integrations/light/#default-turn-on-values

Is there a simpler, easier and quicker way to do that?

Personally, I would set my "_transition duration_" parameter to "_1s_" all the time (if possible: hardcoded accepted on my side), but I can imagine another user would like to set it differently...

thanks

**Comment by rezzalex** [comment]
Hello @gizmocuz , did you have time to see my last post ?

**Comment by rezzalex** [comment]
Hello @gizmocuz , did you have time to see my post last 10/06 ?

**Comment by gizmocuz** [comment]
I'm a bit busy... but parameters could be set in zwavejs2mqtt/zigbee2mqtt and later in domoticz via the hardware setup

**Comment by robertsLando** [comment]
zwave-js-ui now :)

**Comment by waltervl** [comment]
Is this now solved with the Autodiscover number devices configuration? https://www.domoticz.com/wiki/MQTT#MQTT_Autodiscovery_Setup

**Comment by rezzalex** [comment]
I did not noticed this quite recent feature : "_Autodiscover number devices configuration_"
It does't seems to work on my side.... There is no  "discovered configuration devices" on my screen .... :-(
Do I need to force a rediscover from ZwaveJsUI ?

**Comment by waltervl** [comment]
>  There is no "discovered configuration devices" on my screen .... :-( Do I need to force a rediscover from ZwaveJsUI ?

You could give it a try.....

**Comment by rezzalex** [comment]
no, can't see anything in the he Autodiscover number devices configuration. Not sure in which way it is supposed to help...

**Comment by gizmocuz** [comment]
> no, can't see anything in the he Autodiscover number devices configuration. Not sure in which way it is supposed to help...

You mean you pressed the setup button (for the MQTT Auto Discovery hardware) and you see no number configuration options here?

![image](https://github.com/domoticz/domoticz/assets/2265299/16b44463-0fa7-46f5-94dc-b277319f2d4a)

![image](https://github.com/domoticz/domoticz/assets/2265299/10bef378-5c73-42be-b34b-a1e39eb4211c)

If there should be number configuration options, could you post the config topic + payload?

**Comment by rezzalex** [comment]
> You mean you pressed the setup button (for the MQTT Auto Discovery hardware) and you see no number configuration options here?

Yes, nothing

> If there should be number configuration options, could you post the config topic + payload?

I am not sure what should be there, config topic + Payload ? I have 20 or 21 zwave devices managed by ZwaveJsUi... what would you like to see ?

**Comment by gizmocuz** [comment]
Use the application MQTT Explorer
Navigate to zwavejs2mqtt auto discovery topic where you should see a structure like
- sensor
- binary_sensor
- switch
- ...
- number

underneath the number you should have some entries, I would like to have some of these, for example

homeassistant/number/0xb4e3f9fffe02ded3/duration/config

`{"availability":[{"topic":"zigbee2mqtt/bridge/state"}],"command_topic":"zigbee2mqtt/SireneTest/set/duration","device":{"identifiers":["zigbee2mqtt_0xb4e3f9fffe02ded3"],"manufacturer":"Neo","model":"Temperature & humidity sensor and alarm (NAS-AB02B0)","name":"SireneTest"},"entity_category":"config","icon":"mdi:timer","json_attributes_topic":"zigbee2mqtt/SireneTest","max":1800,"min":0,"name":"SireneTest duration","state_topic":"zigbee2mqtt/SireneTest","unique_id":"0xb4e3f9fffe02ded3_duration_zigbee2mqtt","unit_of_measurement":"s","value_template":"{{ value_json.duration }}"}
`

**Comment by rezzalex** [comment]
I will be able to send that early next week, starting from tuesday, dont' close the ticket please :-)

**Comment by rezzalex** [comment]
1st example :

![image](https://github.com/domoticz/domoticz/assets/35136439/6700fbcd-7fce-4f9b-9162-0bcb1701d84a)

{"command_topic":"zwave/Salon/Eclairage_SAlon/38/0/targetValue/set","state_topic":"zwave/Salon/Eclairage_SAlon/38/0/currentValue","state_value_template":"{{ \"OFF\" if value_json.value == 0 else \"ON\" }}","brightness_command_topic":"zwave/Salon/Eclairage_SAlon/38/0/targetValue/set","brightness_scale":99,"brightness_state_topic":"zwave/Salon/Eclairage_SAlon/38/0/currentValue","brightness_value_template":"{{ value_json.value }}","on_command_type":"brightness","device":{"identifiers":["zwavejs2mqtt_0xe573130e_node42"],"manufacturer":"Qubino","model":"Mini Dimmer (ZMNHHD)","name":"Salon-Eclairage SAlon","sw_version":"21.6"},"name":"Salon-Eclairage SAlon_dimmer","unique_id":"zwavejs2mqtt_0xe573130e_42-38-0-currentValue"}


2nd example :  

{"command_topic":"zwave/Salle_à_manger/Eclairage_SAM/38/0/targetValue/set","state_topic":"zwave/Salle_à_manger/Eclairage_SAM/38/0/currentValue","state_value_template":"{{ \"OFF\" if value_json.value == 0 else \"ON\" }}","brightness_command_topic":"zwave/Salle_à_manger/Eclairage_SAM/38/0/targetValue/set","brightness_scale":99,"brightness_state_topic":"zwave/Salle_à_manger/Eclairage_SAM/38/0/currentValue","brightness_value_template":"{{ value_json.value }}","on_command_type":"brightness","device":{"identifiers":["zwavejs2mqtt_0xe573130e_node43"],"manufacturer":"Qubino","model":"DIN Rail Dimmer (ZMNHSD)","name":"Salle à manger-Eclairage SAM","sw_version":"3.1"},"name":"Salle à manger-Eclairage SAM_dimmer","unique_id":"zwavejs2mqtt_0xe573130e_43-38-0-currentValue"}

3rd example : 
![image](https://github.com/domoticz/domoticz/assets/35136439/b3f644f6-9d66-4a3e-8b8d-67e38c37cc41)

{"payload_on":true,"payload_off":false,"value_template":"{{ value_json.value }}","state_topic":"zwave/Chambre_Amis/multi_capteur_fenêtre_ch_amis/48/0/Any","json_attributes_topic":"zwave/Chambre_Amis/multi_capteur_fenêtre_ch_amis/48/0/Any","device":{"identifiers":["zwavejs2mqtt_0xe573130e_node53"],"manufacturer":"Philio Technology Corp","model":"Slim Multi-Sensor (PIR/Door/Temperature/Illumination) (PST02A)","name":"Chambre Amis-multi capteur fenêtre ch amis","sw_version":"1.16"},"name":"Chambre Amis-multi capteur fenêtre ch amis_any","unique_id":"zwavejs2mqtt_0xe573130e_53-48-0-Any"}


I don't remember why I have those 2 different prefix, 1 set for auto-discovery, the other set here in ZwavejsUI : 
![image](https://github.com/domoticz/domoticz/assets/35136439/2980352c-8b50-4bdd-9e55-bfb045c2c209)

![image](https://github.com/domoticz/domoticz/assets/35136439/d55e62e5-92d4-43d0-b0c2-63afc5a8ed1a)


in DZ, the prefix set for autodiscovery is : "homeassistant"

**Comment by gizmocuz** [comment]
Those are not number component types... above in your screenshots I see component types like light/binary_sensor, but no 'number'
I was hoping that for your case, the duration was a configuration property that was available as a 'number' component.

Can you have a look with MQTT Explorer what is send to MQTT when you issue a command with a transition?

zwave/Salon/Eclairage_SAlon/38/0/targetValue/set

(maybe with zwavejs or home assistant)

It might look like

```
{
  "value": 100,
  "options": {
    "transitionDuration": "5s"
  }
}
```

There is no way to configure the nodes default transition time in the zwavejs web interface? (like a one time action)?

**Comment by rezzalex** [comment]
with DZ : 
![image](https://github.com/domoticz/domoticz/assets/35136439/5397f72c-ee43-4a7b-9e58-7687fa5e474d)


with ZwajeUI, no update on any duration parameter, it stays the same, even after changing the duration in ZwaveJsUI:
![image](https://github.com/domoticz/domoticz/assets/35136439/4909ea9d-5963-46fd-ac73-148af5444ba9)
 


> There is no way to configure the nodes default transition time in the zwavejs web interface? (like a one time action)?

Already described by @robertsLando last Mar 4, 2022 in this same ticket : (https://github.com/domoticz/domoticz/issues/5179#issuecomment-1059135079)

**Comment by gizmocuz** [comment]
Please send me a message on slack mentioning this topic as this issue thread is going nowhere and my first reply to this issue still is still the question (but then using transitionDuration)
Better to have a quick chat as you are using zwavejs

**Comment by gizmocuz** [comment]
Not sure if this is solved. I think it is best to solved this in zwavejs-ui as not to add a 'special' option just for one device.
